pkg/prometheus/promcfg.go

```diff
@@ -1767,7 +1780,7 @@ func (cg *ConfigGenerator) generateAlertmanagerConfig(alerting *monitoringv1.Ale
 		// config as well, make sure to path the right namespace here.
 		cfg = addTLStoYaml(cfg, "", am.TLSConfig)
 
-		cfg = append(cfg, cg.generateK8SSDConfig(monitoringv1.NamespaceSelector{}, am.Namespace, apiserverConfig, store, kubernetesSDRoleEndpoint, nil))
+		cfg = append(cfg, cg.generateK8SSDConfig(monitoringv1.NamespaceSelector{}, *am.Namespace, apiserverConfig, store, kubernetesSDRoleEndpoint, nil))
```

Comment:
(suggestion) use ptr.Deref() to return a default value if the pointer is nil
```suggestion
		cfg = append(cfg, cg.generateK8SSDConfig(monitoringv1.NamespaceSelector{}, ptr.Deref(am.Namespace, cg.prom.GetObjectMeta().GetName()), apiserverConfig, store, kubernetesSDRoleEndpoint, nil))
```